src/Reflection/Adapter/ReflectionUnionType.php

```diff
@@ -17,11 +17,11 @@ public function __construct(private BetterReflectionUnionType $betterReflectionT
     }
 
     /**
-     * @return array<ReflectionNamedType|ReflectionType|ReflectionUnionType|null>
+     * @return array<ReflectionNamedType|ReflectionUnionType|null>
      */
     public function getTypes(): array
     {
-        return array_map(static fn (BetterReflectionType $type) => ReflectionType::fromTypeOrNull($type), $this->betterReflectionType->getTypes());
+        return array_map(static fn(BetterReflectionNamedType|BetterReflectionUnionType $type): ReflectionNamedType|ReflectionUnionType|null => ReflectionType::fromTypeOrNull($type), $this->betterReflectionType->getTypes());
```

Comment:
I think this can be simplified to `array_map([ReflectionType::class, 'fromTypeOrNull'], $this->betterReflectionType->getTypes())`